File: mycode/AllocateArrays.m
fprintf('Allocating arrays for transition\n');

it = 1:Ttransition;

if SolveStickyPriceTransition
    ifs = 1;
    irfstruct(ifs).solnSTICKY(it) = struct;
    if DoPriceExperiments
        irfpriceexp.solnSTICKY(it) = struct;
    end
    if SaveCumPolicyFnsIRF
        irfstruct(ifs).cumSTICKY = struct;
        if DoPriceExperiments
            irfpriceexp.cumSTICKY = struct;
        end
    end
end

fprintf('Finished allocating arrays\n')

File: Fortran/AllocateArrays.f90
SUBROUTINE AllocateArrays

USE Parameters
USE Globals
USE Procedures

IMPLICIT NONE

INTEGER	:: it,ifs

write(*,*) 'Allocating arrays for transition'

DO it = 1,Ttransition
 	CALL AllocateSolutionType(solnTRANS(it))
END DO

DO it = 1,Ttransition
	ifs = 0
	IF(SolveStickyPriceTransition==1) CALL AllocateSolutionType(irfstruct(ifs)%solnSTICKY(it))

	IF (DoPriceExperiments==1) THEN
		IF(SolveStickyPriceTransition==1) CALL AllocateSolutionType(irfpriceexp%solnSTICKY(it))
	END IF
	
END DO	

ifs = 0
IF (SaveCumPolicyFnsIRF==1)	THEN
	IF(SolveStickyPriceTransition==1) CALL AllocateCumulativePolicyType(irfstruct(ifs)%cumSTICKY)

	IF (DoPriceExperiments==1) THEN
		IF(SolveStickyPriceTransition==1) CALL AllocateCumulativePolicyType(irfpriceexp%cumSTICKY)
	END IF
END IF	


write(*,*) 'Finished allocating arrays'

END SUBROUTINE